sdk/react/src/markers/MarkerProgressBarOverlay/MarkerProgressBarOverlay.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+
+import styles from './MarkerProgressBarOverlay.styles';
+import type { Marker as MarkerType } from '../../types/Marker';
+
+type Props = {
+  accentColor?: ?string,
+  duration: number,
+  marker: MarkerType,
+};
+
+const MarkerProgressBarOverlay = ({ accentColor, duration, marker }: Props) => {
+  // Prevent rendering overlays for inappropriately configured markers.
+  if ((marker.type === 'text' && !marker.text)
+    || (marker.type === 'icon' && !marker.iconUrl && !marker.imageUrl)) {
+    return null;
+  }
+
+  const style = {};
+
+  style.left = (marker.start === 'start' ? 0 : `${marker.start / duration * 100}%`);
+
+  if (marker.end) {
+    style.right = (marker.end === 'end' ? 0 : `${100 - marker.end / duration * 100}%`);
+  }
+
+  if (marker.markerColor || accentColor) {
```

Comment:
I'm not sure, do we need this checking?

Reply:
From the component perspective it doesn't know anything about persistence of accentColor, because it's passed not as a config, but as a prop, so it can be used without accentColor at all. At the same time markerColor is not required property of the marker, so we need to check if anything of it is present, markerColor has more priority in that case.